usr/lib/cca_stdll/cca_specific.c

```diff
@@ -3321,6 +3321,65 @@ CK_RV token_specific_set_attrs_for_new_object(STDLL_TokData_t *tokdata,
 
     return ret;
 }
+
+/*
+ * ECC private key section (X'20'), Key-usage and translation control flag.
+ */
+#define CCA_XPRTCPAC                          0x01
+#define CCA_ECC_TOKEN_KEYUSAGE_OFFSET         16
+
+static CK_BBOOL ccatok_token_is_cpacf_exportable(const CK_BYTE *token)
+{
+    CK_BYTE keyusage = token[CCA_ECC_TOKEN_KEYUSAGE_OFFSET];
+
+    if (keyusage & CCA_XPRTCPAC)
+        return CK_TRUE;
+
+    return CK_FALSE;
+}
+
+static CK_RV ccatok_pkey_check_attrs(STDLL_TokData_t *tokdata,
+                                     TEMPLATE * templ,
+                                     enum cca_token_type token_type)
```

Comment:
It does not make much sense that the function obtains the key blob internally, but expects the caller to pass the cca_token_type .... The cca_token_type can only be known by the caller if it has obtained the key blob already and passed it through analyse_cca_key_token().

Either: pass blob and cca_token_type  as parameters, or let  the function obtain everything internally.
I would vote for passing everything as paranmeters.

Reply:
Hm, yes, I think the idea was to pass the template, because the function also has to get the CKA_IBM_PROTKEY_EXTRACTABLE attribute internally. So we need the template inside ccatok_pkey_check_attrs anyway. Consequently, the outer routine may also provide CKA_IBM_PROTKEY_EXTRACTABLE as a parm. What do you think?

Reply:
Well, since the caller does not really need CKA_IBM_PROTKEY_EXTRACTABLE, it might not be intuitiv to let it get that attribute and pass it as param. In contrast to that, the blob is anyway obtained by the caller, so here it is quite intuitive to pass it along.  

So I guess I would still pass the template to ccatok_pkey_check_attrs() so that it can get the CKA_IBM_PROTKEY_EXTRACTABLE itself, but pass the blob to it explicitly, since it is already available at the caller. 

Although this is still a mixture if pass and get internally, I guess this is the easiest way for the caller (you have at least 2 callers, maybe more in the future).

Reply:
OK, done.